python/shotgun_fields/list_widget.py

```diff
@@ -8,17 +8,73 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
-"""
-Widget that represents the value of a list field in Shotgun
-"""
+import sgtk
+from sgtk.platform.qt import QtGui, QtCore
 from .label_base_widget import LabelBaseWidget
 from .shotgun_field_meta import ShotgunFieldMeta
 
+shotgun_globals = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_globals")
+
 
 class ListWidget(LabelBaseWidget):
     """
-    Inherited from a :class:`~LabelBaseWidget`, this class is able to
-    display a list field value as returned by the Shotgun API.
+    Display a ``list`` field value as returned by the Shotgun API.
+    """
+    __metaclass__ = ShotgunFieldMeta
+    _DISPLAY_TYPE = "list"
+
+
+class ListEditorWidget(QtGui.QComboBox):
+    """
+    Allows editing of a ``list`` field value as returned by the Shotgun API.
     """
     __metaclass__ = ShotgunFieldMeta
-    _FIELD_TYPE = "list"
+    _EDITOR_TYPE = "list"
+    _IMMEDIATE_APPLY = True
+
+    def get_value(self):
+        """
+        :return: The internal value being displayed by the widget.
+        """
+        return self.currentText()
+
+    def setup_widget(self):
+        """
+        Prepare the widget for display.
+
+        Called by the metaclass during initialization. Adds the valid values to
+        the list and connects the ``activated`` signal.
+        """
+        self.addItem("")
+        valid_values = shotgun_globals.get_valid_values(self._entity_type, self._field_name)
+        self.addItems(valid_values)
+
+        self.activated.connect(
+            lambda i: self.value_changed.emit()
```

Comment:
the `activated` signal sends an index argument which we want to ignore since emit does not accept an argument. we can get at it via the `currentIndex` method later.